File: persiandatepicker/src/main/java/ir/hamsaa/persiandatepicker/api/PersianPickerDate.java
package ir.hamsaa.persiandatepicker.api;

import java.util.Date;

public interface PersianPickerDate {

    void setDate(Long timestamp);

    //The input function takes a unixtime
    void setDateByUnix(Long UnixTimestamp);

    void setDate(Date date);

    void setDate(int persianYear, int persianMonth, int persianDay);

    int getPersianYear();

    int getPersianMonth();

    int getPersianDay();

    int getGregorianYear();

    int getGregorianMonth();

    int getGregorianDay();

    int getDayOfWeek();

    String getPersianMonthName();

    String getPersianDayOfWeekName();

    /**
     * @return String of Persian Date ex: دوشنبه ۱۳ خرداد ۱۳۷۰
     */
    String getPersianLongDate();

    Date getGregorianDate();

    //The function returns a unixtime
    long getTimestamp();



    long getTimeUnix();

    boolean isLeapYear();

}

File: persiandatepicker/src/main/java/ir/hamsaa/persiandatepicker/api/PersianPickerDate.kt
package ir.hamsaa.persiandatepicker.api

import java.util.*

public interface PersianPickerDate {

    fun setDate(timestamp: Long)

    fun setDate(date: Date)

    fun setDate(persianYear: Int, persianMonth: Int, persianDay: Int)

    fun getPersianYear(): Int

    fun getPersianMonth(): Int

    fun getPersianDay(): Int

    fun getGregorianYear(): Int

    fun getGregorianMonth(): Int

    fun getGregorianDay(): Int

    fun getDayOfWeek(): Int

    fun getPersianMonthName(): String

    fun getPersianDayOfWeekName(): String

    /**
     * @return String of Persian Date ex: دوشنبه ۱۳ خرداد ۱۳۷۰
     */
    fun getPersianLongDate(): String

    fun getGregorianDate(): Date

    fun getTimestamp(): Long
}